File: jscomp/test/key_word_property.ml
(* [@@@bs.config {flags = [|
  "-bs-package-output"; "es6:jscomp/test"
|]}] 
*)
(* FIXME it does not work*)


type t 


external default :   t = "default" [@@bs.module "some-es6-module"]
external default2 :   t = "default2" [@@bs.module "some-es6-module"]
let default,default2  = default, default2


external oefault :   t = "default" [@@bs.module "./ome-es6-module"]
external oefault2 :   t = "default2" [@@bs.module "./ome-es6-module"]
let oefault,oefault2  = oefault, oefault2


type window
external window : window = "window" [@@bs.val] [@@bs.module "vscode"]

let window = window
let mk window default = [%obj{window; default ; }]
type t_ = { window : int ; default : int }

let mk2 window default = [{window; default ; }]

let des v = [%obj{window = v##window ; default = v##default }]


let case = 3

let test =  [%obj{case ; window = 3}]

external switch : window -> string = "switch" [@@bs.send]

let u () = switch window




  (* 0,0,0,0,0,0,0,0,0,0,0,0,0,0 *)

File: jscomp/test/key_word_property.js
'use strict';

var Vscode = require("vscode");
var SomeEs6Module = require("some-es6-module");
var SomeEs6Module$1 = require("some-es6-module").default;
var OmeEs6Module = require("./ome-es6-module").default;
var OmeEs6Module$1 = require("./ome-es6-module");

var $$default = SomeEs6Module$1;

var default2 = SomeEs6Module.default2;

var oefault = OmeEs6Module;

var oefault2 = OmeEs6Module$1.default2;

var $$window = Vscode.window;

function mk($$window, $$default) {
  return {
          window: $$window,
          default: $$default
        };
}

function mk2($$window, $$default) {
  return {
          hd: {
            window: $$window,
            default: $$default
          },
          tl: /* [] */0
        };
}

function des(v) {
  return {
          window: v.window,
          default: v.default
        };
}

var test = {
  case: 3,
  window: 3
};

function u(param) {
  return $$window.switch();
}

var $$case = 3;

exports.$$default = $$default;
exports.default = $$default;
exports.__esModule = true;
exports.default2 = default2;
exports.oefault = oefault;
exports.oefault2 = oefault2;
exports.$$window = $$window;
exports.mk = mk;
exports.mk2 = mk2;
exports.des = des;
exports.$$case = $$case;
exports.test = test;
exports.u = u;
/* default Not a pure module */
